### ISSUE TITLE
怎么使用多卡进行推理呢

### ISSUE THREAD

**Issue by linguoqi** [modelscope/swift#428]
使用下面的代码，怎么打开多卡推理的功能呢
`import os
os.environ['CUDA_VISIBLE_DEVICES'] = '4,5,6,7'
import torch
import conf as co
from swift.llm import (
    DatasetName, InferArguments, ModelType, SftArguments
)
from swift.llm import (
    ModelType, get_vllm_engine, get_default_template_type,
    get_template, inference_vllm, inference_stream_vllm
)
from swift.tuners import Swift
from swift.llm import infer_main, sft_main, app_ui_main
import gradio as gr
from typing import Any, Dict, List, Literal, Optional, Tuple, Union
from swift.llm.utils import (History, InferArguments, inference_stream, limit_history_length)

model_type = ModelType.llama2_7b_chat
from swift.llm import (
    get_model_tokenizer, get_template, inference, ModelType, get_default_template_type, inference_stream
)

template_type = get_default_template_type(model_type)

llm_engine = get_vllm_engine(model_type, model_dir='/tf/python/output/llama2-7b-chat/v6-20240205-075623/checkpoint-21000-merged')
# llm_engine = get_vllm_engine(model_type, model_dir=co.best_ckpt_dir)
# template = get_template(template_type, llm_engine.tokenizer)
template = get_template(template_type, llm_engine.hf_tokenizer)
llm_engine.generation_config.max_new_tokens = 2048

request_list = [{'query': 'how to compute opf'}, {'query': 'who are u?'}]
resp_list = inference_vllm(llm_engine, template, request_list)
for request, resp in zip(request_list, resp_list):
    print(f"query: {request['query']}")
    print(f"response: {resp['response']}")
def clear_session() -> History:
    return []

def llamma_bot(content):
    return content

def gradio_chat_demo():
    def model_chat(query: str, history: History) -> Tuple[str, History]:
        old_history, history = limit_history_length(template, query, history, 2048)
        gen = inference_stream_vllm(llm_engine, template,
                                    [{
                                        'query': query,
                                        'history': history
                                    }])
        for resp_list in gen:
            history = resp_list[0]['history']
            total_history = old_history + history
            yield '', total_history

    model_name = model_type
    with gr.Blocks() as demo:
        gr.Markdown(f'<center><font size=8>{model_name} Bot</center>')

        chatbot = gr.Chatbot(label=f'{model_name}')
        message = gr.Textbox(lines=2, label='Input')
        with gr.Row():
            clear_history = gr.Button('🧹 清除历史对话')
            send = gr.Button('🚀 发送')
        send.click(
            model_chat, inputs=[message, chatbot], outputs=[message, chatbot])
        clear_history.click(
            fn=clear_session, inputs=[], outputs=[chatbot], queue=False)
    demo.queue().launch(root_path="/7b-app", height=1000, share=False, server_port=8083, server_name='0.0.0.0')
    return demo
demo = gradio_chat_demo()
`

**Comment by Jintao-Huang** [comment]
请设置`tensor_parallel_size`, 传入get_vllm_engine
